addons/cert.py

```diff
@@ -335,6 +335,30 @@ def str11(data):
         if valueToken.isNumber and int(valueToken.str)==strlen:
             reportError(valueToken, 'style', 'Do not specify the bound of a character array initialized with a string literal', 'STR11-C')
 
+# API01-C
+# Avoid laying out strings in memory directly before sensitive data
+def api01(data):
+        for scope in data.scopes:
+            if scope.type!='Struct':
```

Comment:
ok...one question regarding Python support: at the moment I'm using Python 2.7 but the support will end end of this year. Does the Python interface of Cppcheck already support Python 3.x?

Reply:
The addon scripts itself should work with Python 2.7 and 3.x. At least it is intended to work. They get compiled with both versions during the Travis check. But this catches only syntax errors. Some tests are also executed with both versions if i remember it correctly. Is that what you mean? I am not sure what you mean with "Python interface of Cppcheck" exactly 😀
Hope this helps a bit.
When working on Cppcheck scripts i try to test it with both versions.

The only script that does not work with Python 3 that i have knowledge of is tools/donate-cpu-server.py. But it only runs on the server so it is not that important. But it should be upgraded (danmar wrote somewhere that it would be good to make it work with Python 3, Python 2 compatibility is not necessary, if i remember it correctly).